### PR TITLE
Add GitHub action to merge develop into list-command daily

### PR DESCRIPTION
**Pull request by andrew-zhan139**
#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [x] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [x] Write/update integration tests
- [x] Write/update functional tests if needed
- [x] `make pr` passes
- [x] `make update-reproducible-reqs` if dependencies were changed
- [x] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by mildaniel — APPROVED** [review]

File: .github/workflows/merge-develop-into-list-command.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  schedule:
+    - cron: '* 1 * * *'
+jobs:
+  update-branch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Merge develop -> list-command
+        uses: actions/checkout@v3
+      - run: |
+          curl -X POST -H "Accept: application/vnd.github.v3+json" -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" https://api.github.com/repos/aws/aws-sam-cli/merges -d '{"base":"list-command","head":"develop","commit_message":"Merge develop into list-command"}'
```
Comment (hawflau) [inline .github/workflows/merge-develop-into-list-command.yml:13]: nit: you can also just use Github CLI to do the same. - https://cli.github.com/manual/gh_api

File: .github/workflows/merge-develop-into-list-command.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: Merge the develop branch into the list-command branch
+
+on:
+  schedule:
+    - cron: '* 1 * * *'
```
Comment (hawflau) [inline .github/workflows/merge-develop-into-list-command.yml:5]: IIRC GitHub Actions only run automatically on default branch (i.e. develop branch). Will this schedule work?

**Review by hawflau — COMMENTED** [review]
Looks okay overall. I just have two questions (see individual comments). Also, are we going to revert this PR when we are ready to merge the `list-command` branch to `develop` branch?

**Comment by mildaniel** [comment]
This needs to be merged to develop. Let's also add a comment documenting the fact that this GH action should be removed after merging this feature to develop.

**Pull request closed without merging**